### PR TITLE
[@types/leaflet] Update ImageOverlay options and add setZIndex method

### PR DESCRIPTION
**Pull request by hwu1001**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

Documentation for the options:
https://leafletjs.com/reference-1.4.0.html#imageoverlay-crossorigin
https://leafletjs.com/reference-1.4.0.html#imageoverlay-erroroverlayurl
https://leafletjs.com/reference-1.4.0.html#imageoverlay-zindex

For the method:
https://leafletjs.com/reference-1.4.0.html#imageoverlay-setzindex

- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

I didn't see any previous tests for `ImageOverlay`'s methods so I added the others as well. If those aren't supposed to be included I can remove them and just leave the options.

### CODE REVIEW

File: types/leaflet/index.d.ts [hunk 1]
```diff
@@ -549,12 +549,16 @@ export namespace tileLayer {
     function wms(baseUrl: string, options?: WMSOptions): TileLayer.WMS;
 }
 
+export type CrossOrigin = boolean | string;
```
Comment (atd-schubert) [inline types/leaflet/index.d.ts]: Can you apply the CrossOrigin type also to the other occurrences?

**Review by atd-schubert — COMMENTED** [review]
Please apply the CrossOrigin type to the other properties. Everything else seems to be fine for me...

File: types/leaflet/index.d.ts (hunk 1 above)
Comment (v1zix) [reply types/leaflet/index.d.ts]: Yep, no problem. The only other place I could find it in the Leaflet documentation was in `TileLayer`:
https://leafletjs.com/reference-1.4.0.html#tilelayer-crossorigin

I've updated that interface now.

Comment (v1zix) [reply types/leaflet/index.d.ts]: @atd-schubert Does this PR look okay now?

File: types/leaflet/index.d.ts [hunk 2]
```diff
@@ -550,12 +550,16 @@ export namespace tileLayer {
...
```
Comment (minestarks) [inline types/leaflet/index.d.ts]: any reason to `export` the type?

Comment (v1zix) [reply types/leaflet/index.d.ts]: I was under the impression that all types should to be exported even if it seems unnecessary based on this documentation: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md. If I run `npm run lint leaflet` after making the change to this:
```ts
type CrossOrigin = boolean | string;
```

I get the following output:
```
$ npm run lint leaflet

> definitely-typed@0.0.3 lint /my/path/Documents/dev/DefinitelyTyped
> dtslint types "leaflet"

Error: /my/path/Documents/dev/DefinitelyTyped/types/leaflet/index.d.ts:552:6
ERROR: 552:6  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md

    at /my/path/Documents/dev/DefinitelyTyped/node_modules/dtslint/bin/index.js:160:19
    at Generator.next (<anonymous>)
    at fulfilled (/my/path/Documents/dev/DefinitelyTyped/node_modules/dtslint/bin/index.js:5:58)
    at <anonymous>
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! definitely-typed@0.0.3 lint: `dtslint types "leaflet"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the definitely-typed@0.0.3 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /my/path/.npm/_logs/2019-02-13T01_15_08_050Z-debug.log
```

Based on the error I think I would have to turn automatic exporting off for the whole module, which seems like it should be discussed in a separate issue.

Comment (minestarks) [reply types/leaflet/index.d.ts]: My bad, didn’t catch that requirement!

**Pull request merged**